kedro/config/omegaconf_config.py

```diff
@@ -275,7 +277,9 @@ def load_and_merge_dir_config(
             return {}
         if len(aggregate_config) == 1:
             return list(aggregate_config)[0]
-        return dict(OmegaConf.merge(*aggregate_config))
+        return OmegaConf.to_container(
+            OmegaConf.merge(*aggregate_config, self.runtime_params), resolve=True
```

Comment:
Does this definitely work when `self.runtime_params` is a dictionary and not a `DictConfig`? i.e. is it ok to not do `OmegaConf.create(runtime_params)` here? Looking through omegaconf source I'm pretty sure this is fine (only the first argument of `OmegaConf.merge` needs to be `DictConfig`) but please do check this!

Reply:
I checked with the source code, `merge` can take any primitive type  I think it's fine here.

```python
    def merge(
        *configs: Union[
            DictConfig,
            ListConfig,
            Dict[DictKeyType, Any],
            List[Any],
            Tuple[Any, ...],
            Any,
        ]
```